spec/models/tribunal_case/tribunal_case_purge_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'spec_helper'
+
+RSpec.describe TribunalCase, '.purge!', type: :model do
+  subject { described_class }
+  let(:cases) { class_double(described_class) }
+
+  before do
+    allow(subject).to receive(:where).and_return(cases)
+    allow(cases).to receive(:delete_all)
+    subject.purge!('2017-01-01')
+  end
+
+  it 'finds cases created before a specified date' do
+    expect(subject).to have_received(:where).with(['created_at < ?', '2017-01-01'])
```

Comment:
I don't think so.  I'd have historically done exactly that, but  I was going for full isolation as test suite speed is becoming an issue.

This way ensures the ORM receives the necessary scoping query without having to touch the DB.  I'm increasingly reluctant to persist test records if I don't need to.  My reasoning is: 

1) We're already concerned about the length of time it takes to complete a full test run; 
1) That models have a larger impact on test suite speed as they undergo mutation testing;
1) We have a stated ambition of increasing mutation testing so that it covers the entire app;
1) We're not going to change the ORM for the foreseeable future;
1) It seems highly unlikely we'll change the *nature* of the scope of the `WHERE` clause (but it's easy enough to change the spec if we do);
1) The two-record approach feels a bit like testing the ORM (although I'm happy to do that where speed isn't becoming an issue).